ipam/allocator.go

```diff
@@ -122,19 +112,68 @@ func (a *Allocator) updateBitMasks(aSpace *addrSpace) error {
 func (a *Allocator) checkConsistency(as string) {
 	// Retrieve this address space's configuration and bitmasks from the datastore
 	a.refresh(as)
+	a.Lock()
 	aSpace, ok := a.addrSpaces[as]
+	a.Unlock()
 	if !ok {
 		return
 	}
 	a.updateBitMasks(aSpace)
+	aSpace.Lock()
```

Comment:
Right, the reason I am locking is in fact because I do not want to make any assumptions.
I am missing what problem would cause locking the address space while running `CheckConsistency` on the bitmask. An address space is tied to a datastore, we need to lock it while we run the consistency check on its subnets.